account_invoice_inter_company/tests/test_inter_company_invoice.py

```diff
@@ -14,44 +14,591 @@ class TestAccountInvoiceInterCompanyBase(SavepointCase):
+                "company_id": cls.company_b.id,
+            }
+        )
+
         cls.account_obj = cls.env["account.account"]
         cls.account_move_obj = cls.env["account.move"]
+        cls.product_consultant_multi_company = cls.env[
+            "product.product"
+        ].create(
+            {
+                "name": "Service Multi Company",
+                "uom_id": cls.env.ref("uom.product_uom_hour").id,
+                "uom_po_id": cls.env.ref("uom.product_uom_hour").id,
+                "categ_id": cls.env.ref("product.product_category_3").id,
+                "type": "service",
+            }
+        )
         # if product_multi_company is installed
+
         if "company_ids" in cls.env["product.template"]._fields:
             # We have to do that because the default method added a company
-            cls.env.ref(
-                cls.module + ".product_consultant_multi_company"
-            ).company_ids = False
-        cls.invoice_company_a = cls.env.ref(cls.module + ".customer_invoice_company_a")
-        cls.user_company_a = cls.env.ref(cls.module + ".user_company_a")
-        cls.user_company_b = cls.env.ref(cls.module + ".user_company_b")
-        cls.company_a = cls.env.ref("account_invoice_inter_company.company_a")
-        cls.company_b = cls.env.ref("account_invoice_inter_company.company_b")
+            cls.product_consultant_multi_company.company_ids = False
+
+        cls.env["ir.sequence"].create(
+            {
+                "name": "Account Sales Journal Company A",
+                "prefix": "SAJ-A/%(year)s/",
+                "company_id": cls.company_a.id,
+            }
+        )
+        cls.pcg_X58 = cls.env["account.account.template"].create(
+            {
+                "name": "Internal Transfers",
+                "code": "X58",
+                "user_type_id": cls.env.ref(
+                    "account.data_account_type_current_assets"
+                ).id,
+                "reconcile": True,
+            }
+        )
+
+        cls.a_recv_company_a = cls.env["account.account"].create(
+            {
+                "code": "X11002-A",
+                "name": "Debtors - (company A)",
+                "internal_type": "receivable",
+                "reconcile": "True",
+                "user_type_id": cls.env.ref(
+                    "account.data_account_type_receivable"
+                ).id,
+                "company_id": cls.company_a.id,
+            }
+        )
+        cls.a_pay_company_a = cls.env["account.account"].create(
+            {
+                "code": "X1111-A",
+                "name": "Creditors - (company A)",
+                "internal_type": "payable",
+                "reconcile": "True",
+                "user_type_id": cls.env.ref(
+                    "account.data_account_type_payable"
+                ).id,
+                "company_id": cls.company_a.id,
+            }
+        )
+        cls.invoice_company_a = cls.env["account.move"].create(
```

Comment:
Seems this line is failing:

`
 Remaining fields to compute before setting up registry: defaultdict(<class 'set'>, {account.move.amount_tax_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_total_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_residual: {<NewId 0x7f33b94a96d8>}, account.move.invoice_payment_state: {<NewId 0x7f33b94a96d8>}, account.move.amount_total: {<NewId 0x7f33b94a96d8>}, account.move.amount_residual_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_untaxed_signed: {<NewId 0x7f33b94a96d8>}, account.move.amount_untaxed: {<NewId 0x7f33b94a96d8>}, account.move.amount_tax: {<NewId 0x7f33b94a96d8>}, account.move.line.account_root_id: {<NewId 0x7f33baf29978>}, account.move.line.account_internal_type: {<NewId 0x7f33baf29978>}, account.move.line.amount_residual_currency: {<NewId 0x7f33baf29978>}, account.move.line.reconciled: {<NewId 0x7f33baf29978>}, account.move.line.amount_residual: {<NewId 0x7f33baf29978>}, account.move.line.tax_audit: {<NewId 0x7f33baf29978>}})
`